client/types/util.d.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file Misc. type declarations utilized across the repo.
+ */
+
+// TODO: determine if this file & folder is a good idea
+
+export declare global {
+  /** Represents any function, without using the {@link Function} object. */
+  type AnyFunction<T = unknown> = (...args: any) => T
```

Comment:
I think there's a convention to use `R` for return types?
```suggestion
  type AnyFunction<R = unknown> = (...args: any) => R
```

Reply:
There is no specific convention, and usually if it's just one type variable you just name it `T`. I don't really care though?